src/hotspot/share/gc/shenandoah/shenandoahVerifier.cpp

```diff
@@ -604,7 +604,7 @@ class VerifyThreadGCState : public ThreadClosure {
 public:
   VerifyThreadGCState(const char* label, char expected) : _label(label), _expected(expected) {}
   void do_thread(Thread* t) {
-    char actual = ShenandoahThreadLocalData::gc_state(t);
+    char actual = ShenandoahThreadLocalData::gc_state(t) & ~((char)ShenandoahHeap::WEAK_ROOTS);
```

Comment:
Why these adjustments are needed? I think it shows we don't play well with GC-state lifecycle here...

Reply:
The verifier would complain about the extra bit being set or maybe not set (e.g. concurrent cycles would set it, but degen cycles would not). We haven't verified conc-weakroots-in-progress before, so I figured it would be easiest to keep it that way and mask it out.

Reply:
ShenandoahGCStateResetter used to disable the flag, so it makes sense.